Gordon360/ApiControllers/HousingController.cs

```diff
@@ -170,14 +172,21 @@ public IHttpActionResult SaveApplication([FromBody] ApartmentApplicationViewMode
                 throw new BadInputException() { ExceptionMessage = errors };
             }
             //get token data from context, username is the username of current logged in person
-            ClaimsPrincipal authenticatedUser = this.ActionContext.RequestContext.Principal as ClaimsPrincipal;
+            ClaimsPrincipal authenticatedUser = ActionContext.RequestContext.Principal as ClaimsPrincipal;
             string username = authenticatedUser.Claims.FirstOrDefault(x => x.Type == "user_name").Value;
 
             string sessionID = Helpers.GetCurrentSession().SessionCode;
 
-            string editorUsername = applicationDetails.EditorUsername;
+            string editorUsername = applicationDetails?.EditorProfile?.AD_Username ?? applicationDetails?.EditorUsername;
 
             ApartmentApplicantViewModel[] apartmentApplicants = applicationDetails.Applicants;
+            foreach (ApartmentApplicantViewModel applicant in apartmentApplicants)
+            {
+                if (applicant.Profile == null)
+                {
+                    applicant.Profile = _profileService.GetStudentProfileByUsername(applicant?.Profile?.AD_Username ?? applicant?.Username);
```

Comment:
We know that `applicant.Profile` is null in this context, so checking it for `AD_Username` is redundant.
```suggestion
                    applicant.Profile = _profileService.GetStudentProfileByUsername(applicant?.Username);
```

Reply:
Perhaps you should double check that `applicant.Username` has a value in this context, because otherwise this is going to fail.

Reply:
Oh right this happened because it used to be used to get a property before the applicant.Profile thing was built in, so it was something like `_profileService.GetStudentProfileByUsername(stuff).SomeProperty`

Basically I guess I didn't think to double check the insides of that for redundancy,

Reply:
If both Profile and Username are null, then it should stop and throw an error, though I think in the near future this loop won't be needed, since in theory it should be setup so that it is impossible for the frontend to send it would a Profile object.
It is sort-of a backwards compatibility thing that will be changed/removed once we are sure the newer frontend stuff is doing what it is intended to do

Reply:
```suggestion
                    applicant.Profile = _profileService.GetStudentProfileByUsername(applicant.Username);
```